### PR TITLE
Give pserve the ability top open server in browser

### PR DESCRIPTION
**Pull request by msabramo**
E.g.:

    pserve app.ini --browser

or:

    pserve app.ini -b

Cc: @sontek, @mmerickel, @mcdonc

### CODE REVIEW

File: pyramid/scripts/pserve.py [hunk 1]
```diff
@@ -334,6 +341,15 @@ def serve():
...
                 self.out('Exiting%s (-v to see traceback)' % msg)
 
+        if self.options.webbrowser_open:
+            def open_browser():
+                context = loadcontext(SERVER, app_spec, name=app_name, relative_to=base,
+                        global_conf=vars)
+                url = 'http://{host}:{port}/'.format(**context.config())
+                time.sleep(1)
+                webbrowser.open(url)
+            threading.Thread(target=open_browser).start()
```
Comment (mmerickel) [inline pyramid/scripts/pserve.py]: can we mark the thread as detached so that it doesn't block the process from terminating?

Comment (msabramo) [inline pyramid/scripts/pserve.py]: Done.

**Comment by mmerickel** [comment]
I'm amazed to see that the `run` method has `no cover` on it. We need to fix that in the future. Anyway if you could add a changelog entry I'll merge this.

**Comment by mmerickel** [comment]
I'm going back and forth on the cli option names `-w` and `--webbrowser-open`. Is there any other tool that provides similar options? I'd lean toward like `--open-browser` or `-w` myself.

**Comment by msabramo** [comment]
```
$ gem server --help
...
    -l, --launch[=COMMAND]           launches a browser window
                                     COMMAND defaults to 'start' on Windows
                                     and 'open' on all other platforms
```

**Comment by msabramo** [comment]
```
$ ipython notebook --help
...
--browser=<Unicode> (NotebookApp.browser)
    Default: u''
    Specify what command to use to invoke a web browser when opening the
    notebook. If not specified, the default browser will be determined by the
    `webbrowser` standard library module, which allows setting of the BROWSER
    environment variable to override it.
...
```

**Comment by msabramo** [comment]
Maybe `-b` or `--browser`?

**Comment by mmerickel** [comment]
Yeah I can't think of something that `-b` would clobber right now, so it's fine with me.

**Comment by msabramo** [comment]
Cool, updated.

**Comment by mmerickel** [comment]
Awesome! Thanks!

**Comment by marc1n** [comment]
This change does not respect --server-name option. It only works when server name is 'main' (default).

**Comment by mmerickel** [comment]
@marc1n Sure, if you would like to submit a PR fixing this it'd be accepted.

**Pull request merged**